### PR TITLE
Include 'git status' as part of system diagnostics when debugging a pipeline

### PR DESCRIPTION
**Pull request by Roman-Shchukin**
**Issue description:**
Currently doesn't outline the status of git repository while running pipeline

**Fix description:**
Added calls of git status command in the checkout step of the pipeline when Enable system diagnostics switch on.

image

**Changes:**
-  `GitStatus` method with call git status command added in `Agent.Plugins.GitCliManager` and `Agent.Worker.Build.GitCommandManager` classes
- Calls of `GitStatus` method added in `Agent.Plugins.GitSourceProvider` and `Agent.Worker.Build.GitSourceProvider` respectively

**Documentation changes required:** No

**Added unit tests:** No

### CODE REVIEW

File: src/Agent.Worker/Build/GitSourceProvider.cs [hunk 1]
```diff
@@ -1038,6 +1042,21 @@ public abstract class GitSourceProvider : SourceProvider, ISourceProvider
                     IOUtil.DeleteFile(_clientCertPrivateKeyAskPassFile);
                 }
             }
+
+            await RunGitStatusIfSystemDebug(executionContext, targetPath);
+        }
+
+        private async Task RunGitStatusIfSystemDebug(IExecutionContext executionContext, string targetPath)
```
[the remark this answers is not in the document]
Comment (Roman-Shchukin) [reply src/Agent.Worker/Build/GitSourceProvider.cs:1049]: It is made to reduce code duplication. It calls three times.

**Review by kirill-ivlev — APPROVED** [review]

**Review by ismayilov-ismayil — APPROVED** [review]

**Review by mmrazik — APPROVED** [review]

**Comment by Roman-Shchukin** [comment]
We added a call to the git status command in the checkout step of the pipeline when Enable system diagnostics switch on.
Pull request ([link](https://github.com/microsoft/azure-pipelines-agent/pull/3947)) successfully merged and is waiting rolled out,
We will update this ticket accordingly when release will hit ring 5.

**Pull request merged**